DisguiseUnityRenderStream/Runtime/DisguiseRenderStream.cs

```diff
@@ -432,6 +396,92 @@ protected void AwaitFrame()
             DisguiseFramerateManager.Update();
         }
 
+        // Updates the RenderTextures assigned to image parameters on the render thread to avoid stalling the main thread
+        protected void OnBeginContextRendering(ScriptableRenderContext context, List<Camera> cameras)
+        {
+            if (!HasNewFrameData)
+                return;
+            
+            if (LatestFrameData.scene >= m_Schema.scenes.Length)
+                return;
+        
+            var spec = m_Schema.scenes[LatestFrameData.scene];
+            var images = m_SceneFields[LatestFrameData.scene].images;
+            if (images == null)
+                return;
+
+            // Only run once per frame for the main render context
+            if (cameras.Any(x => x.cameraType != CameraType.Game))
+                return;
+            
+            var nImageParameters = spec.parameters.Count(t => t.type == RemoteParameterType.RS_PARAMETER_IMAGE);
+
+            using var imageData = new NativeArray<ImageFrameData>(nImageParameters, Allocator.Temp);
+            if (PluginEntry.instance.GetFrameImageData(spec.hash, imageData.AsSpan()) != RS_ERROR.RS_ERROR_SUCCESS)
+                return;
+            
+            while (m_ScratchTextures.Count < imageData.Length)
+            {
+                var index = m_ScratchTextures.Count;
+                
+                m_ScratchTextures.Add(new Texture2D(
+                    (int)imageData[index].width,
+                    (int)imageData[index].height,
+                    PluginEntry.ToTextureFormat(imageData[index].format),
+                    false,
+                    true));
+            }
+
+            var i = 0;
+            foreach (var field in images)
+            {
+                if (field.GetValue() is RenderTexture renderTexture)
+                {
+                    Texture2D texture = m_ScratchTextures[i];
+                    if (texture.width != imageData[i].width ||
+                        texture.height != imageData[i].height ||
+                        texture.format != PluginEntry.ToTextureFormat(imageData[i].format))
+                    {
+                        m_ScratchTextures[i] = new Texture2D(
```

Comment:
Temporary RTs could be a great fit for the scratch textures (for texture re-use + automatic lifetime management). Working on it.

Reply:
The code for temp RTs is in place: https://github.com/wenzhang-unity/RenderStream-Unity/tree/mark/input-temporary-rt

But RTs can only be typeless which isn't supported by [CUDA interop](https://docs.nvidia.com/cuda/cuda-runtime-api/group__CUDART__D3D11.html#group__CUDART__D3D11_1g85d07753780643584b8febab0370623b) down the line. Disguise reports:

```
GPGPUSystem: Failed to register DX11 texture with CUDA: 1
Invalid value ID3D11Texture2D of format w:299 h:290 - DXGI_FORMAT_R32G32B32A32_TYPELESS. MipLevels: 1, Sample Desc: 1 samples. Usage D3D11_USAGE_DEFAULT
RenderStream Callstack:  < Cuda::registerDXTexture < Core_Impl::copyImageFrame < RenderStream::Core::getImageFrameParameter < rs_getFrameImage < RSGetFrameImage
```

Sticking to Texture2D for now, maybe with our own pooling implementation.

Reply:
This is good information. Can we capture this somewhere in the inline docs?